pkg/controller/elasticsearch/migration/migrate_data.go

```diff
@@ -24,68 +24,24 @@ const (
 	AllocationExcludeAnnotationName = "elasticsearch.k8s.elastic.co/allocation-exclude"
 )
 
-func shardIsMigrating(toMigrate client.Shard, others []client.Shard) bool {
-	if toMigrate.IsRelocating() || toMigrate.IsInitializing() {
-		return true // being migrated away or weirdly just initializing
-	}
-	if !toMigrate.IsStarted() {
-		return false // early return as we are interested only in started shards for migration purposes
-	}
-	for _, otherCopy := range others {
-		if otherCopy.IsStarted() {
-			return false // found another shard copy
-		}
-	}
-	return true // we assume other copies are initializing or there are no other copies
-}
-
-// nodeIsMigratingData is the core of IsMigratingData just with any I/O
-// removed to facilitate testing. See IsMigratingData for a high-level description.
-func nodeIsMigratingData(nodeName string, shards client.Shards, exclusions map[string]struct{}) bool {
-	// all other shards not living on the node that is about to go away mapped to their corresponding shard keys
-	othersByShard := make(map[string][]client.Shard)
-	// all shard copies currently living on the node leaving the cluster
-	candidates := make([]client.Shard, 0)
-
-	// divide all shards into the to groups: migration candidate or other shard copy
-	for _, shard := range shards {
-		_, ignore := exclusions[shard.NodeName]
-		if shard.NodeName == nodeName {
-			candidates = append(candidates, shard)
-		} else if !ignore {
-			key := shard.Key()
-			others, found := othersByShard[key]
-			if !found {
-				othersByShard[key] = []client.Shard{shard}
-			} else {
-				othersByShard[key] = append(others, shard)
-			}
-		}
-	}
-
-	// check if there is at least one shard on this node that is migrating or needs to migrate
-	for _, toMigrate := range candidates {
-		if shardIsMigrating(toMigrate, othersByShard[toMigrate.Key()]) {
-			return true
-		}
-	}
-	return false
-
-}
-
 // IsMigratingData looks only at the presence of shards on a given node
 // and checks if there is at least one other copy of the shard in the cluster
 // that is started and not relocating.
-func IsMigratingData(ctx context.Context, shardLister esclient.ShardLister, podName string, exclusions []string) (bool, error) {
+func IsMigratingData(ctx context.Context, shardLister esclient.ShardLister, podName string) (bool, error) {
 	shards, err := shardLister.GetShards(ctx)
 	if err != nil {
 		return false, err
 	}
-	excludedNodes := make(map[string]struct{}, len(exclusions))
-	for _, name := range exclusions {
-		excludedNodes[name] = struct{}{}
+	// all shard copies currently living on the node leaving the cluster
+	shardsToMigrate := make([]client.Shard, 0)
+	// filter shards affected by node removal
+	for _, shard := range shards {
+		if shard.NodeName == podName {
+			shardsToMigrate = append(shardsToMigrate, shard)
+		}
 	}
-	return nodeIsMigratingData(podName, shards, excludedNodes), nil
+	return len(shardsToMigrate) > 0, nil
```

Comment:
I think we don't need to maintain a slice. We can return `true` as soon as one shard is hosted on this node.

```golang
	for _, shard := range shards {
		if shard.NodeName == podName {
			return true, nil
		}
	}
	return false, nil
```

Reply:
Absolutely, sloppy me :-(